pkg/authn/delegating.go

```diff
@@ -0,0 +1,28 @@
+package authn
+
+import (	
+	"time"
+	
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authentication/authenticatorfactory"
+	authenticationclient "k8s.io/client-go/kubernetes/typed/authentication/v1beta1"
+)
+
+// NewDelegatingAuthenticator creates an authenticator compatible with the kubelet's needs
+func NewDelegatingAuthenticator(client authenticationclient.TokenReviewInterface, authn *AuthnConfig) (authenticator.Request, error) {
+
+	authenticatorConfig := authenticatorfactory.DelegatingAuthenticatorConfig{
+		Anonymous:    false, // always require authentication
+		CacheTTL:     2 * time.Minute,
+		ClientCAFile: authn.X509.ClientCAFile,
+		TokenAccessReviewClient: client,
+	}
+
+	// if client == nil {
+	// 	return nil, errors.New("no client provided, cannot use webhook authentication")
+	// }
+	// authenticatorConfig.TokenAccessReviewClient = client
```

Comment:
Let's remove commented code.

Reply:
Yes, my mistake, forgot to uncomment... I have moved the check to the beginning of func